plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php

```diff
@@ -117,14 +118,45 @@ public function is_complete() {
 			$third_party_complete = apply_filters( 'woocommerce_admin_third_party_tax_setup_complete', false );
 
 			$this->is_complete_result = $is_wc_connect_taxes_enabled ||
-				count( TaxDataStore::get_taxes( array() ) ) > 0 ||
-				get_option( 'woocommerce_no_sales_tax' ) !== false ||
-				$third_party_complete;
+				false !== get_option( 'woocommerce_no_sales_tax' ) ||
+				$third_party_complete ||
+				$this->is_actioned();
+
+			if ( ! $this->is_complete_result && $this->has_existing_tax_rates() ) {
+				$this->mark_actioned();
+			}
 		}
 
 		return $this->is_complete_result;
 	}
 
+	/**
+	 * Determines if a tax rate exists in the database.  Result is indefinitely cached.
+	 *
+	 * @return bool
+	 */
+	private function has_existing_tax_rates() {
+		global $wpdb;
+
+		$has_existing_tax_rates = wp_cache_get( 'woocommerce_onboarding_task_tax_rates_exist' );
+		if ( false === $has_existing_tax_rates ) {
+			$rate_exists            = (bool) $wpdb->get_var( "SELECT 1 {$wpdb->prefix}woocommerce_tax_rates limit 1" );
```

Comment:
This query will always return `1`; we'll need a `FROM` in here:

```suggestion
			$rate_exists            = (bool) $wpdb->get_var( "SELECT 1 FROM {$wpdb->prefix}woocommerce_tax_rates limit 1" );
```